### DISCUSSION TITLE
Result for aggregation function, e.g. row count

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#76] [Q&A] [answered]
Will we support to have value returned within a result set?

[UPDATE] I found that metrics is not the aggregation value...

![image](https://github.com/user-attachments/assets/7ee08b83-3696-4ea9-aa9c-b8c8f717d90a)

![image](https://github.com/user-attachments/assets/7549e9d9-ff7b-4a94-a3a1-613a4faf5e25)

![image](https://github.com/user-attachments/assets/15bc87e2-d083-4109-940f-d2139de9559b)

Related test case:
[count test](https://github.com/NTDLS/NTDLS.Katzebase.Server/blob/experiment_select_count_result/TestApplications/NTDLS.Katzebase.UnitTests/Scripts/Engine/DDLExecution/BasicTest.DDLExecution.ver.0.0.1.fsx)

**Comment by ingted** [comment]
Commit d293eba5237f87573a0c5722ab2fc9f6f19f0828 resolved this.

**Comment by ingted** [comment]
Oops, count(*) is not supported so far...

![image](https://github.com/user-attachments/assets/52a3a295-ca17-4049-a94c-4175828db9a4)

**Reply by ingted** [reply]
count(1) => correct
count(id) => caused error when document doesn't have id field

**Reply by ingted** [reply]
Another issue:
```
        let preparedQueries = StaticQueryParser.ParseBatch(_core, "SELECT COUNT(1) FROM testSch", userParameters)
        let preparedQuery = preparedQueries.Item 0
        
        let queryResultCollection = _core.Query.ExecuteQuery(preLogin, preparedQuery)
        equals 1 queryResultCollection.Collection.Count
        equals 1 queryResultCollection.RowCount
```

queryResultCollection.Collection.Count is correct since the row with count value returned. Collection Count is 1, passed.
Not sure if queryResultCollection.RowCount should be 1 or?

**Reply by ingted** [reply]
Last possible issue(?):

```
if (value == null)
{
    if (property.PropertyType.IsValueType && Nullable.GetUnderlyingType(property.PropertyType) == null)
    {
        continue; // Skip setting value if property is non-nullable value type
    }
    else
    {
        property.SetValue(obj, null);
    }
}
else
{
    property.SetValue(obj, Convert.ChangeType(value, Nullable.GetUnderlyingType(property.PropertyType) ?? property.PropertyType));
}
```

Seems like "continue" caused MapTo<T> a type like
```
class Count {
    int count
}
```

Doesn't work? (need more test...)

**Reply by ingted** [reply]
So far we have this in test case:
```
        let queryResultCollection = _core.Query.ExecuteQuery(preLogin, preparedQuery)
        equals 1 queryResultCollection.Collection.Count

        let queryDocList = ((queryResultCollection.Collection.Item 0) :?> KbQueryDocumentListResult).Rows
        equals 1 queryDocList.Count
        equals "2" queryDocList[0].Values[0] //inserted two documents
```

**Reply by NTDLS** [reply, marked as answer]
MaptTo<> only supports setting properties, not fields so this works:
![image](https://github.com/user-attachments/assets/037d17e9-59d3-47e6-ad21-e51be2f59813)

When grouping, the "affected row count" should return the number of grouped rows, not the total underlying row count, so I thinkk what we have is good.

**Reply by ingted** [reply]
Do (will) we support count(*) or not (will not)?

**Reply by NTDLS** [reply]
We do now! 😃
